MetaMask/metamask-extension
# Deprecate old Popover and Modals

## Explanation
Currently there are multiple version of what would be considered a `Modal` component and it is not clear which engineers should be using. This PR adds a JS Doc deprecation tag to three components `Popover` and 2 `Modal` components that should no longer be used

* Fixes #19218 
* Fixes #18975

## Screenshots/Screencaps



### After
Instances of `ui/components/ui/popover/popover.component.js`

![Screenshot 2023-06-09 at 3 04 32 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/e879ed52-697e-4ed9-b7bb-fd807abff9bb)

Instances of `ui/components/app/modal/modal.component.js`

![Screenshot 2023-06-09 at 3 05 16 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/d7eac494-9c1d-47eb-b726-325a0c92a090)

Instances of `ui/components/app/modal/modal.component.js`

![Screenshot 2023-06-09 at 3 09 18 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/d113fa64-c72a-4f7a-a94f-0a43d34b8327)

## Manual Testing Steps
To see the deprecation indicators make sure your VS code settings are enabled to see deprecation tag support 

- Pull this branch
- Search for an instance of a deprecated component in the code base
- See deprecation message 

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
